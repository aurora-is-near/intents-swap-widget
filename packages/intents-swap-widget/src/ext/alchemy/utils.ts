import type { ValueOf } from '@/types/utils';
import type { Chains } from '@/types/chain';

export const revertAlchemyChainsMap = <
  T extends Partial<Record<Chains, string>>,
>(
  chainsMap: T,
) => {
  return Object.fromEntries(
    Object.entries(chainsMap).map(([key, value]) => [value, key]),
  ) as {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    [V in ValueOf<T> & (string | number | symbol)]: keyof typeof chainsMap;
  };
};
