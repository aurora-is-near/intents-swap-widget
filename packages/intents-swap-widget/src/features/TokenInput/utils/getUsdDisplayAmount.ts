import { formatUsdAmount } from '@/utils/formatters/formatUsdAmount';
import type { Token } from '@/types/token';

export const getUsdDisplayAmount = (
  token: Token,
  value: string,
  quoteUsdValue?: number,
) => {
  if (quoteUsdValue) {
    return formatUsdAmount(quoteUsdValue);
  }

  const valueNum = value && value !== 'NaN' ? parseFloat(value) : 0;

  return valueNum && !Number.isNaN(valueNum)
    ? formatUsdAmount(valueNum * token.price)
    : '$0.00';
};
