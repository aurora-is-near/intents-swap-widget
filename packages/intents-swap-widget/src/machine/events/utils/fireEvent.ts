import { machine } from '@/machine/machine';
import type { TradeEvents } from '@/machine/events';

export const fireEvent = <E extends keyof TradeEvents>(
  event: E,
  payload: TradeEvents[E],
) => {
  machine.fire(event, payload);
};
