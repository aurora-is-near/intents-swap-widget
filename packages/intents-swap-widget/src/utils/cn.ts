import clsx from 'clsx';
import { extendTailwindMerge } from 'tailwind-merge';
import type { ClassValue } from 'clsx';

const twMerge = extendTailwindMerge({
  extend: {
    classGroups: {
      'font-size': [
        'text-sw-label-s',
        'text-sw-label-m',
        'text-sw-label-l',
        'text-sw-h1',
        'text-sw-h2',
        'text-sw-h3',
        'text-sw-h4',
        'text-sw-p-s',
        'text-sw-p-m',
        'text-sw-display',
      ],
    },
  },
});

export const cn = (...inputs: ClassValue[]) => twMerge(clsx(inputs));
