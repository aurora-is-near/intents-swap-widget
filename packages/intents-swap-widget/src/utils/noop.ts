export const noop = () => {
  // This function does nothing
};
