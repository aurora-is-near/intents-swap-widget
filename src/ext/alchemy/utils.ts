import type { ValueOf } from '@/types/utils';
import type { Chains } from '@/types/chain';

export const revertAlchemyChainsMap = <
  T extends Partial<Record<Chains, string>>,
>(
  chainsMap: T,
) => {
  return Object.fromEntries(
    Object.entries(chainsMap).map(([key, value]) => [value, key]),
  ) as {
    [V in ValueOf<T> & (string | number | symbol)]: keyof typeof chainsMap;
  };
};
