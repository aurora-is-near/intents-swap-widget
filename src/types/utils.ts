export type ValueOf<T> = NonNullable<T[keyof T]>;

export type DeepReadonly<T> =
  T extends Map<infer K, infer V>
    ? ReadonlyMap<DeepReadonly<K>, DeepReadonly<V>>
    : T extends Set<infer S>
      ? ReadonlySet<DeepReadonly<S>>
      : T extends object
        ? { readonly [K in keyof T]: DeepReadonly<T[K]> }
        : T;
