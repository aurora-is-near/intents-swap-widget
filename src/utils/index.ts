export { isDebug } from './checkers/isDebug';
export { isNotEmptyAmount } from './checkers/isNotEmptyAmount';
export { isValidChain } from './checkers/isValidChain';

export { isEth } from './evm/isEth';
export { isEvmChain } from './evm/isEvmChain';
export { isEvmToken } from './evm/isEvmToken';
export { isEvmBaseToken } from './evm/isEvmBaseToken';
export { getEvmTokenBalance } from './evm/getEvmTokenBalance';
export { getEvmMainTokenBalance } from './evm/getEvmMainTokenBalance';

export { formatBigToHuman } from './formatters/formatBigToHuman';
export { formatHumanToBig } from './formatters/formatHumanToBig';
export { formatTinyNumber } from './formatters/formatTinyNumber';
export { formatTxHash } from './formatters/formatTxHash';
export { formatUsdAmount } from './formatters/formatUsdAmount';
export { getExplorerUrl } from './formatters/getExplorerUrl';
export { getTransactionLink } from './formatters/getTransactionLink';

export { cn } from './cn';
export { noop } from './noop';
export { notReachable } from './notReachable';
