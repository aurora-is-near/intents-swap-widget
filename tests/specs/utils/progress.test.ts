import { getStreamsProgress } from '../../../src/utils/progress';
import { streamsSchedule } from '../../fixtures/streams-schedule';

jest.useFakeTimers().setSystemTime(new Date('2024-11-12T00:00:00Z'));

describe('Progress', () => {
  describe('getStreamsProgress', () => {
    it('returns the progress of each stream', () => {
      expect(getStreamsProgress(streamsSchedule)).toMatchSnapshot();
    });
  });
});
