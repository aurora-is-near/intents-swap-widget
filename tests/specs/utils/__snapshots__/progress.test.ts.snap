// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress getStreamsProgress returns the progress of each stream 1`] = `
[
  49.82118399026023,
  248.95437262357416,
  248.95437262357416,
  248.95437262357416,
  995.8174904942966,
  159.90424325672453,
]
`;
